channels/tests/test_worker.py

```diff
@@ -35,3 +55,49 @@ def test_channel_filters(self):
             worker.apply_channel_filters(["yes.1", "no.1", "maybe.2", "yes"]),
             ["yes.1"],
         )
+
+    def test_run_with_consume_later_error(self):
+
+        # consumer with ConsumeLater error at first call
+        def _consumer(message, **kwargs):
+            if not hasattr(_consumer, '_called'):
+                _consumer._called = True
+                raise ConsumeLater()
+
+        consumer = mock.Mock(side_effect=_consumer)
+        chenels = ['test', ]
+        channel_layer = mock.MagicMock()
+        channel_layer.router.channels = chenels
+        channel_layer.receive_many = mock.MagicMock(return_value=('test', {}))
```

Comment:
You don't need to mock this; the ChannelTestCase base class provides a pre-made in-memory channel layer you can prep messages on and extract sent messages from in a much less fragile way. See https://github.com/andrewgodwin/channels/blob/master/channels/tests/test_request.py#L20